gto-support-androidx-viewpager2/src/main/java/org/ccci/gto/android/common/androidx/viewpager2/widget/ViewPager2.kt

```diff
@@ -0,0 +1,16 @@
+package org.ccci.gto.android.common.androidx.viewpager2.widget
+
+import androidx.recyclerview.widget.RecyclerView
+import androidx.viewpager2.widget.ViewPager2
+
+/**
+ * Provides a work-around for https://issuetracker.google.com/issues/181785654
+ */
+fun ViewPager2.whileMaintainingVisibleCurrentItem(block: (RecyclerView.Adapter<*>?) -> Unit) {
+    val adapter = adapter
+    val visible = adapter?.takeIf { it.hasStableIds() }?.getItemId(currentItem)
```

Comment:
When would a page not have ```stableIds```?

Reply:
stable ids is a concept that starts at the underlying `RecyclerView`. Think about rendering a list of Strings that could have duplicates. There is no way to have a unique id for each string because of the duplicates unless you track additional meta-data with those strings.

without a unique/stable id we have no way of tracking where a string moves to if we reorder the list.

So, specifying an adapter has stable ids will indicate that each item will have an id that is unique to it and won't change as you are updating the underlying data.

Ideally whenever we create an adapter we define a mechanism to provide stable ids, but because this is more generic re-usable logic we are accounting for adapters that don't have stable ids.